### PR TITLE
Avoid playwright's buggy test.fail() skipping tests

### PR DESCRIPTION
**Pull request by lukaw3d**
A lot of playwright tests on master are being skipped, and CI doesn't fail
https://github.com/oasisprotocol/oasis-wallet-web/actions/runs/5826329626/job/15800051758#step:10:19
> image

This has been happening since https://github.com/oasisprotocol/oasis-wallet-web/pull/1570.
Additionally, https://github.com/oasisprotocol/oasis-wallet-web/pull/1583/commits/84b30f3c5270843d525ae8bb83dcea518fe572ad broke a test, and it wasn't caught.

Skipped tests seem to be an upstream issue. Reported to: https://github.com/microsoft/playwright/issues/26435

### CODE REVIEW

File: playwright/tests/fiat.spec.ts [hunk 1]
```diff
@@ -113,12 +126,11 @@
...
-      expect(permissionsPolicy.find(rule => rule.startsWith(`${permission}=`))).toContain(
-        'https://global.transak.com',
-      )
-      expect(permissionsPolicy.find(rule => rule.startsWith(`${permission}=`))).toContain(
-        'https://global-stg.transak.com',
-      )
+      const rule = permissionsPolicy.find(rule => rule.startsWith(`${permission}=`))!
+      expect(
+        rule.endsWith('=*') ||
+          (rule.includes('https://global.transak.com') && rule.includes('https://global-stg.transak.com')),
```
[the remark this answers is not in the document]
Comment (pro-wh) [reply playwright/tests/fiat.spec.ts:132]: rule is not a single URL right? should we suppress this?

**Review by pro-wh — COMMENTED** [review]
those later tests are now run, but can this still report failing tests?

image

image

File: playwright/tests/fiat.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lukaw3d) [reply playwright/tests/fiat.spec.ts:132]: Yeah, already suppressed to the best of my abilities. It's also just a test, testing our non-evil code :shrug:

**Comment by doperiddle** [comment]
Yes

**Comment by lukaw3d** [comment]
Steps in https://github.com/oasisprotocol/oasis-wallet-web/pull/1588/commits/41da30749eebd7b333ec1e6da805915e0ffcf16d and https://github.com/oasisprotocol/oasis-wallet-web/pull/1588/commits/14f52305643d7ab01169a265b9ac7c0c88353de3
- split `beforeEach` into `expectNoErrorsInConsole(page); setup(page);`
- removed `expectNoErrorsInConsole` in tests with `test.fail`:
  Playwright skips when combining `test.fail` and assertions in event listeners (in `expectNoErrorsInConsole` there's: `page.on('pageerror', error => { expect(` and `page.on('console', message => { ... expect(`)
- added `page.waitForRequest` that gets blocked by security headers.
  Instead of finding errors in console.
- (when playwright is fixed we can re-add expectNoErrorsInConsole)

**Pull request merged**